examples/point_discharge/point_discharge.py

```diff
@@ -0,0 +1,261 @@
+        0.23069756 + y1*(0.3488590e-1 + y1*(0.262698e-2 + y1*(0.10750e-3 + y1*0.74e-5))))))
+    y2 = 2.0/x
+    expr2 = exp(-x)/sqrt(x)*(1.25331414 + y2*(-0.7832358e-1 + y2*(0.2189568e-1 + y2*(
+        -0.1062446e-1 + y2*(0.587872e-2 + y2*(-0.251540e-2 + y2*0.53208e-3))))))
+    return conditional(ge(x, 2), expr2, expr1)
+
+
+class PointDischargeParameters(object):
+    """
+    Problem parameter class, including point source representation.
+
+    Delta functions are not in H1 and hence do not live in the FunctionSpaces we
+    seek to use. Here we use a Gaussian approximation with a small radius. The
+    small radius has been calibrated against the analytical solution. See [2]
+    for details.
+    """
+    def __init__(self, offset):
+        self.offset = offset
+
+        # Physical parameters
+        self.diffusivity = Constant(0.1)
+        self.viscosity = None
+        self.drag = Constant(0.0025)
+        self.uv = Constant(as_vector([1.0, 0.0]))
+        self.elev = Constant(0.0)
+
+        # Stabilisation
+        self.use_lax_friedrichs_tracer = True
+
+        # Parametrisation of point source
+        self.source_x, self.source_y = 2.0, 5.0
+        self.source_r = 0.05606298 if self.use_lax_friedrichs_tracer else 0.05606298
+        self.source_value = 100.0
+
+        # Boundary conditions
+        self.boundary_conditions = {
+            'tracer': {
+                1: {'value': Constant(0.0)},      # inflow
+                2: {'open': None},                # outflow
+                3: {'diff_flux': Constant(0.0)},  # Neumann
+                4: {'diff_flux': Constant(0.0)},  # Neumann
+            },
+            'shallow_water': {
+                1: {
+                    'uv': Constant(as_vector([1.0, 0.0])),
+                    'elev': Constant(0.0)
+                },                                # inflow
+                2: {
+                    'uv': Constant(as_vector([1.0, 0.0])),
+                    'elev': Constant(0.0)
+                },                                # outflow
+                3: {'un': Constant(0.0)},         # free-slip
+                4: {'un': Constant(0.0)},         # free-slip
+            }
+        }
+
+    def ball(self, mesh, triple, scaling=1.0, eps=1.0e-10):
+        x, y = SpatialCoordinate(mesh)
+        expr = lt((x-triple[0])**2 + (y-triple[1])**2, triple[2]**2 + eps)
+        return conditional(expr, scaling, 0.0)
+
+    def gaussian(self, mesh, triple, scaling=1.0):
+        x, y = SpatialCoordinate(mesh)
+        expr = exp(-((x-triple[0])**2 + (y-triple[1])**2)/triple[2]**2)
+        return scaling*expr
+
+    def source(self, fs):
+        triple = (self.source_x, self.source_y, self.source_r)
+        return self.gaussian(fs.mesh(), triple, scaling=self.source_value)
+
+    def bathymetry(self, fs):
+        return Function(fs).assign(5.0)
+
+    def quantity_of_interest_kernel(self, mesh):
+        triple = (20.0, 7.5 if self.offset else 5.0, 0.5)
```

Comment:
minor detail: took me a while to figure out what triple is, you could define a x,y,radius dict and pass **dict to ball.